io/zenoh-transport/src/unicast/lowlatency/link.rs

```diff
@@ -14,29 +14,25 @@
 use super::transport::TransportUnicastLowlatency;
 #[cfg(feature = "stats")]
 use crate::stats::TransportStats;
-use crate::TransportExecutor;
+use crate::{unicast::link::TransportLinkUnicast, TransportExecutor};
 use async_std::task;
 use async_std::{prelude::FutureExt, sync::RwLock};
-use zenoh_codec::*;
-use zenoh_core::{zasyncread, zasyncwrite};
-
 use std::sync::Arc;
 use std::time::Duration;
 use zenoh_buffers::{writer::HasWriter, ZSlice};
-use zenoh_link::LinkUnicast;
-use zenoh_protocol::transport::{
-    BatchSize, KeepAlive, TransportBodyLowLatency, TransportMessageLowLatency,
-};
+use zenoh_codec::*;
+use zenoh_core::{zasyncread, zasyncwrite};
+use zenoh_protocol::transport::{KeepAlive, TransportBodyLowLatency, TransportMessageLowLatency};
 use zenoh_result::{zerror, ZResult};
 use zenoh_sync::RecyclingObjectPool;
 
 pub(crate) async fn send_with_link(
-    link: &LinkUnicast,
+    link: &TransportLinkUnicast,
```

Comment:
Because `TransportLinkUnicast` contains the actual `LinkUnicast` negotiated configuration. E.g.: agreed batch size, compression (even if may not be wise to enable it), etc. When doing a `send` or `recv` with a `TransportLinkUnicast` all those parameters are taken into account. With `LinkUnicast` they are not.

Reply:
Ok, I've got the point!

Reply:
As additional clarification, `TransportLinkUnicast` now does not include any queue. It's a wrapper around `LinkUnicast` that takes into account all negotiated parameters.